mem.go

```diff
@@ -198,18 +198,19 @@ func (m *MemoryHandler) Find(ctx context.Context, lookup *resource.Lookup, page,
 		}
 		// Apply pagination
 		total := len(items)
-		start := (page - 1) * perPage
 		end := total
-		if perPage > 0 {
-			end = start + perPage
+		start := offset
+
+		if limit > 0 {
+			end = start + limit
 			if start > total-1 {
 				start = 0
 				end = 0
 			} else if end > total-1 {
 				end = total
 			}
 		}
-		list = &resource.ItemList{total, page, items[start:end]}
+		list = &resource.ItemList{Total: total, Page: offset, Skip: offset, Items: items[start:end]}
```

Comment:
It's still wrong, the `page` is not the `offset` and the `skip` is not the `offset` either.

In this setup, the `offset` is computed from the `page` and `skip` API parameter. This means the storage engine is no longer able to know the value of those parameters, and thus can't init those value in the `ItemList` type. I think those two values (`Page` and `Skip`) should now be set by the caller (rest-layer `resource` package).

Does it make sense?